packages/rules/src/mustSatisfyPeerDependencies.ts

```diff
@@ -71,11 +71,19 @@ export const MATCH_ANY_VERSION_RANGE = /^(\*|x)$/;
  *
  * More info: https://docs.npmjs.com/about-semantic-versioning
  */
-export const MATCH_MAJOR_VERSION_RANGE = /^(\^?\d+|\^?\d+\.x|\^?\d+\.x\.x|\^\d+\.\d+|\^\d+\.\d+\.x|\^\d+\.\d+\.\d+)$/;
+export const MATCH_MAJOR_VERSION_RANGE = /^(\^?\d+|\^?\d+\.x|\^?\d+\.x\.x|\^\d+\.\d+|\^\d+\.\d+\.x|\^\d+\.\d+\.\d+(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?)$/;
```

Comment:
e.g. the first `\d+` could accidentally allow a leading zero, but I didn't want to make this:
```suggestion
export const MATCH_MAJOR_VERSION_RANGE = /^(\^?(?:0|[1-9]\d*)|\^?(?:0|[1-9]\d*)\.x|\^?(?:0|[1-9]\d*)\.x\.x|\^(?:0|[1-9]\d*)\.(?:0|[1-9]\d*)|\^(?:0|[1-9]\d*)\.(?:0|[1-9]\d*)\.x|\^(?:0|[1-9]\d*)\.(?:0|[1-9]\d*)\.(?:0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?)$/;
```
(and the same for `RANGE_REGEX`, too)